### PR TITLE
Add countResponseFields function

### PR DESCRIPTION
**Pull request by soluchok**
**Stack**:
- #99
- #98 ⬅
- #93

⚠️ *Part of a stack created by [spr](https://github.com/ejoffe/spr). Do not merge manually using the UI - doing so may have unexpected results.*

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -628,6 +628,50 @@ function modResponse(response, extended) {
...
+
+  const incr = (key, val) => {
+    if (countResponseFieldsRecursive(hm, key, val)) {
+      return;
+    }
+
+    let current = 0;
+    if (hm[key]) {
+      current = hm[key];
+    }
```
Comment (kolia-kaploniuk) [inline index.js:655]: suggestion: can be shortened to `let current = hm[key] || 0;`

File: index.js [hunk 2]
```diff
@@ -628,6 +628,50 @@ function modResponse(response, extended) {
...
 }
 
+function countResponseFields(resp) {
+  const counts = {};
+
+  if (resp.data) {
+    countResponseFieldsRecursive(counts, "data", resp.data);
+  }
+
+  counts.errors = resp.errors ? resp.errors.length : 0;
```
Comment (kolia-kaploniuk) [inline index.js:638]: is this line needed? In the follow-up PR, errors are passed into the sidecar as they are anyway

File: index.js [hunk 3]
```diff
@@ -628,6 +628,50 @@ function modResponse(response, extended) {
...
+  }
+
+  for (const [k, v] of Object.entries(val)) {
+    incr(`${prefix}.${k}`, v);
+  }
+
+  return false;
+}
+
+exports.countResponseFields = countResponseFields;
```
Comment (kolia-kaploniuk) [inline index.js:674]: is it intended to be used outside of the plugin?

**Review by kolia-kaploniuk — APPROVED** [review]

File: index.js (hunk 3 above)
Comment (soluchok) [reply index.js:690]: it is used for testing

**Comment by soluchok** [comment]
✓ Commit merged in pull request [#99](https://github.com/inigolabs/inigo-js/pull/99)

**Pull request closed without merging**